addons/transifex/models/ir_translation.py

```diff
@@ -8,17 +8,58 @@
     from ConfigParser import ConfigParser
 from os.path import join as opj
 import os
+import re
 import werkzeug
 
-from odoo import models, fields
+from odoo import models, fields, _
 from odoo.modules.module import ad_paths
 
+REGEX_PLACEHOLDERS = re.compile("%(\(\w+\)|)([0+-]|)(\d+|\*|)[diouxXcrsa]|%(\(\w+\)|)([0+-]|)(\d+|\*|)(\.(\d+|\*|)|)[eEfFgG]|%%")
+REGEX_XML_TAGS = re.compile("</?(\w+)")
 
 class IrTranslation(models.Model):
 
     _inherit = 'ir.translation'
 
     transifex_url = fields.Char("Transifex URL", compute='_get_transifex_url')
+    is_valid = fields.Boolean(default=True)
+
+    def _compute_validity(self):
+        bad = self.browse()
+        for translation in self:
+            if not translation.value:
+                continue
+
+            # detect missing %s
+            placeholders_src = sorted([m.group(0) for m in REGEX_PLACEHOLDERS.finditer(translation.src)])
+            placeholders_trans = sorted([m.group(0) for m in REGEX_PLACEHOLDERS.finditer(translation.value)])
+            if placeholders_src != placeholders_trans:
+                bad |= translation
+                continue
+
+            # detect tags mismatches
+            placeholders_src = sorted([m.group(0) for m in REGEX_XML_TAGS.finditer(translation.src)])
+            placeholders_trans = sorted([m.group(0) for m in REGEX_XML_TAGS.finditer(translation.value)])
```

Comment:
findall would be simpler here, also above IFF REGEX_PLACEHOLDERS is modified to use non-capturing groups (findall returns a list of match strings, or group tuples if there are groups).